python/ray/workflow/workflow_executor.py

```diff
@@ -99,10 +102,16 @@ async def run_until_complete(
             ready_futures = await self._poll_ready_tasks()
 
             # ----------- handle ready tasks -----------
-            for fut in ready_futures:
-                await self._handle_ready_task(
-                    fut, workflow_id=workflow_id, wf_store=wf_store
-                )
+            await asyncio.gather(
+                *[
+                    self._handle_ready_task(
+                        fut, workflow_id=workflow_id, wf_store=wf_store
+                    )
+                    for fut in ready_futures
+                ]
+            )
```

Comment:
```suggestion
            await asyncio.gather(
                    self._handle_ready_task(
                        fut, workflow_id=workflow_id, wf_store=wf_store
                    )
                    for fut in ready_futures
            )
```

Reply:
`asyncio.gather` does not accept generators